src/components/PostDetails.tsx

```
@@ -1,152 +1,124 @@
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
+  const postId = post?.id ?? null;
+  const postDetails = post || posts.find(p => p.id === postId);
 
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+  useEffect(() => {
+    if (postId !== null) {
+      dispatch(fetchCommentsByPost(postId));
+      setVisible(false);
     }
-  };
+  }, [dispatch, postId]);
 
-  useEffect(() => {
-    loadComments();
-  }, []);
+  if (!postDetails) {
+    return (
+      <div className="notification is-warning" data-cy="PostDetails">
+        No post selected
+      </div>
+    );
+  }
 
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+  const addComment = async ({
+    name,
+    email,
+    body,
+  }: {
+    name: string;
+    email: string;
+    body: string;
+  }) => {
+    if (postId === null) {
+      return;
+    }
 
-  const addComment = async ({ name, email, body }: CommentData) => {
     try {
-      const newComment = await commentsApi.createComment({
-        name,
-        email,
-        body,
-        postId: post.id,
-      });
-
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
+      await dispatch(postComment({ name, email, body, postId })).unwrap();
     } catch (error) {
```

The `addComment` thunk call is wrapped in try/catch but the catch block is empty. At minimum log the error (`console.error(error)`) or show a notification so failures are observable during development. Silent failures make debugging harder.